caffe2/python/trt/test_pt_onnx_trt.py

```diff
@@ -0,0 +1,255 @@
+            model = models.vgg11(pretrained=True)
+        elif model_name == "vgg11_bn":
+            model = models.vgg11_bn(pretrained=True)
+        elif model_name == "vgg13":
+            model = models.vgg13(pretrained=True)
+        elif model_name == "vgg13_bn":
+            model = models.vgg13_bn(pretrained=True)
+        elif model_name == "vgg16":
+            model = models.vgg16(pretrained=True)
+        elif model_name == "vgg16_bn":
+            model = models.vgg16_bn(pretrained=True)
+        elif model_name == "vgg19":
+            model = models.vgg19(pretrained=True)
+        elif model_name == "vgg19_bn":
+            model = models.vgg19_bn(pretrained=True)
+
+        elif model_name == "densenet121":
+            model = models.densenet121(pretrained=True)
+        elif model_name == "densenet161":
+            model = models.densenet161(pretrained=True)
+        elif model_name == "densenet169":
+            model = models.densenet169(pretrained=True)
+        elif model_name == "densenet201":
+            model = models.densenet201(pretrained=True)
+
+        elif model_name == "alexnet":
+            model = models.alexnet(pretrained=True)
+        else:
+            self.fail("Model {} is not yet supported".format(model_name))
+
+        shape = (1,) + input_shape
+        dummy_input  = (torch.randn(shape),)
+        onnx_name = model_name + ".onnx"
+
+        torch.onnx.export(model,
+                          dummy_input,
+                          onnx_name,
+                          input_names = [],
+                          output_names = [],
+                          verbose=False,
+                          export_params=True,
+                          opset_version=9)
+
+        with self.build_engine_onnx(onnx_name) as engine:
+            h_input, d_input, h_output, d_output, stream = allocate_buffers(engine)
+            with engine.create_execution_context() as context:
+                err_count = 0
+                for index, f in enumerate(self.image_files):
+                    test_case = load_normalized_test_case(input_shape, f,\
+                        h_input, normalization_hint)
+                    cuda.memcpy_htod_async(d_input, h_input, stream)
+
+                    context.execute_async_v2(bindings=[d_input, d_output],
+                                             stream_handle=stream.handle)
+                    cuda.memcpy_dtoh_async(h_output, d_output, stream)
+                    stream.synchronize()
+
+                    amax = np.argmax(h_output)
+                    pred = self.labels[amax]
+                    if "_".join(pred.split()) not in\
+                            os.path.splitext(os.path.basename(test_case))[0]:
+                        err_count = err_count + 1
+                self.assertLessEqual(err_count, 1, "Too many recognition errors")
+
+    def test_alexnet(self):
+        self._test_model("alexnet", (3, 227, 227))
+
+    def test_resnet18(self):
```

Comment:
we can just keep small models like resnet18, squeezenet1_0, and so on, and skip large models to keep the testing within reasonable time.

Reply:
Great, doing that...

Reply:
Less than minute per model left.